### ISSUE TITLE
zip_open reports "No zip archive" even if the zip file seems to be fine

### ISSUE THREAD

**Issue by rzvN3k** [nih-at/libzip#373]
**Describe the Bug**
zip_open returns "No zip archive", but I'm not sure what is the reason for this. I'm new into zip files (compression/decompression methods in general) and I don't know what I miss.

**Expected Behavior**
zip_open should return no error

**To Reproduce**
In my application, I retrieve with cURL library via HTTPS from a server (Spring Boot server) a zip file containing 2 files: 1 key file and 1 cert file. Download via HTTPS of zip file is successful, but I'm struggling to extract the archive.

For opening the zip archive, the code is:
```
        zip* certsArchive = nullptr;
        zip_file* zipFile = nullptr;

        certsArchive = zip_open(zipPath.c_str(), 0, &error);
        if(certsArchive == nullptr)
        {
            zip_error_to_str(errstr, sizeof(errstr), error, errno);
            std::cout << "[SSL Client] Failed to open zip archive with the client cert and key. Error message = " << errstr << '\n';
            result = -1;
        }
        else
        {
            ...
        }

```

Error:

![message_error](https://user-images.githubusercontent.com/39876652/230780177-82680665-3962-49d3-bd9c-7694d33e3f1a.png)

**libzip Version**
libzip-1.9.2

**Operating System**
Ubuntu 20.04

**Additional context**
Unzip tool from Ubuntu is returning nothing suspicious about the target zip file
![image](https://user-images.githubusercontent.com/39876652/230780135-7f9cf096-30d4-4fac-82b1-8932613d8aff.png)

**Comment by 0-wiz-0** [comment]
I see this has been edited. There was a zip archive linked before. I tested that zip archive:
```
ziptool -c response.zip cat 0
```
and the same with `cat 1`, and libzip was happy with the file and provided the output.
Please try a similar test for your actual zip archive.

As I said, the file you originally provided could be read by libzip without problems, even with the additional consistency checks done with `-c`.

**Comment by rzvN3k** [comment]
Hello,

Tested with the commands you provided, and the content is printed without any problems, but the problem persists when I try to open the zip file programmatically with the "zip_open" function, which returns "No zip archive" (piece of the code which do that is above provided").

**Comment by rzvN3k** [comment]
Hi,

I managed to resolve this error. The problem "No zip archive" was reported because I was calling my extraction function BEFORE closing the zip file (it was opened for write operation, to the disk, after it was fetched from the server via HTTPS).

Now, it seems that the files from the zip are extracted, but I am getting a stack error.

![image](https://user-images.githubusercontent.com/39876652/236679605-0dedfda9-d01f-4cfc-a09e-dbada8163ba5.png)

For sure this is also some error logic in my code, I will debug further and I keep you updated

**Comment by rzvN3k** [comment]
Hello again,

Managed to solve the stack smash issue too, the files are extracted properly and ready to be used further in application. 

The problem was that I was calling "fread" function with a size which can be greater than the max size which I declared for the buffer used in the read function (just letting this if anyone is facing a similar issue).

Thanks for your time @0-wiz-0, I think this issue can be closed.
